front/pages/i/_shortId/_slug.vue

```diff
@@ -177,28 +177,32 @@ export default {
   $_veeValidate: {
     validator: 'new'
   },
-  async asyncData({ app, route, store, error }) {
-    const isMyIdea = store.getters['userData/userId'] === route.params.userId
+
+  async asyncData({ app, route, store, error, res }) {
     try {
       const { data } = await app.$amplifyApi.graphql({
-        query: isMyIdea ? getMyIdea : getUsersIdea,
-        variables: {
-          userId: route.params.userId,
-          ideaId: route.params.ideaId
-        },
-        authMode: isMyIdea ? undefined : 'API_KEY'
+        query: getIdea,
+        variables: { shortId: route.params.shortId },
+        authMode: 'API_KEY'
```

Comment:
that remove of myIdea vs getIdea might be not correct. As we will have private ideas that is only visible to user itself, and so if we auth against API key for both user own idea and not own idea we wont be able to not show private ideas.

Reply:
The thing is that url doesn't have user id, just idea id, so there is no way to run getMyIdea. If this is private idea it just should check against currently authenticated user. This have to be then in getIdea mapping template.

Reply:
if auth mode api key we wont be able to get user in template and do any checks at all afaik

Reply:
How then it's supposed to perform owner check in Appsync when items can be either public or private? Is it not possible to perform that same action with authenticated and non authenticated request?

Reply:
If auth is done via api_key its always anonymous, only when authed against cognito pool we will be able to get user and do checks based on it.

Reply:
So then how it's possible to optionally check authenticated user? So if requested idea is public it should be able to get it without authentication and if idea is private then only return it if it's owner, otherwise return 401/403. Only way I can think of is then to do two requests in a row, but this is ineffective.

Reply:
if authMode is undefined it should use cognito pool auth so i guess it should use it if user logged in

Reply:
Ok, I made changes, should be ok now